api/v1beta2/helmrepository_types.go

```diff
@@ -91,6 +92,11 @@ type HelmRepositorySpec struct {
 	// +required
 	Interval metav1.Duration `json:"interval"`
 
+	// Insecure allows connecting to a non-TLS HTTP container registry.
+	// This field is only taken into account if the .spec.type field is set to 'oci'.
+	// +optional
+	Insecure bool `json:"insecure,omitempty"`
```

Comment:
I think we should make this field name more explicit to (1) make it obvious what it does and (2) allow for any future "insecure" flags to be added.
```suggestion
	InsecurePlainHTTP bool `json:"insecure,omitempty"`
```
`Insecure` as a prefix still makes sense to me to make it obvious this is an insecure transport.

Reply:
This field matches what we have in OCIRepository and what's in the RFC.

Reply:
That's unfortunate. I still believe it's a way better UX if it had a more telling name.

Reply:
I pointed that out in the RFC PR, too, by the way: https://github.com/fluxcd/flux2/pull/3081#pullrequestreview-1152887299

Reply:
The goal here at present is to get the feature in while continuing to be uniform. Dealing with the naming of the field would be a separate task, as aligning it across kinds would first require an amendment to the RFC.

Reply:
I'm fine with merging this as my comment on the RFC PR also got completely ignored, anyway.